docs/src/main/asciidoc/hibernate-search-guide.adoc

```diff
@@ -0,0 +1,627 @@
+[source,java]
+----
+package org.acme.hibernate.search.elasticsearch.config;
+
+import org.hibernate.search.backend.elasticsearch.analysis.ElasticsearchAnalysisConfigurer;
+import org.hibernate.search.backend.elasticsearch.analysis.model.dsl.ElasticsearchAnalysisDefinitionContainerContext;
+
+public class AnalysisConfigurer implements ElasticsearchAnalysisConfigurer {
+
+    @Override
+    public void configure(ElasticsearchAnalysisDefinitionContainerContext context) {
+        context.analyzer("defaultAnalyzer").custom() // <1>
+                .withTokenizer("standard")
+                .withTokenFilters("asciifolding", "lowercase");
+
+        context.analyzer("stemmingAnalyzer").custom() // <2>
+                .withTokenizer("standard")
+                .withTokenFilters("asciifolding", "lowercase", "porter_stem");
+
+        context.normalizer("defaultNormalizer").custom() // <3>
+                .withTokenFilters("asciifolding", "lowercase");
+    }
+}
+----
+<1> This is a simple analyzer separating the words on spaces, removing any non-ASCII characters by its ASCII counterpart (and thus removing accents) and putting everything in lowercase.
+It is used in our examples for the author's names.
+<2> We are a bit more aggressive with this one and we include some stemming: we will be able to search for `mystery` and get a result even if the indexed input contains `mysteries`.
+It is definitely too aggressive for person names but it is perfect for the book titles.
+<3> Here is the normalizer used for sorting. Very similar to our first analyzer, except we don't tokenize the words as we want one and only one token.
+
+== Adding full text capabilities to our REST service
+
+In our existing `LibraryResource`, we just need to inject the following methods (and add a few ``import``s):
+
+[source,java]
+----
+    @Transactional // <1>
+    void onStart(@Observes StartupEvent ev) throws InterruptedException { // <2>
+        // only reindex if we imported some content
+        if (Book.count() > 0) {
+            Search.getSearchSession(em)
+                    .createIndexer()
+                    .startAndWait();
+        }
+    }
+
+    @GET
+    @Path("author/search")
+    @Transactional
+    public List<Author> searchAuthors(@QueryParam("pattern") String pattern) { // <3>
+        List<Author> authors = Search.getSearchSession(em) // <4>
+                .search(Author.class) // <5>
+                .predicate(f ->
+                    pattern == null || pattern.trim().isEmpty() ?
+                            f.matchAll() : // <6>
+                            f.simpleQueryString()
+                                .onFields("firstName", "lastName", "books.title").matching(pattern) // <7>
+                )
+                .sort(f -> f.byField("lastName_sort").then().byField("firstName_sort")) // <8>
+                .fetchHits(); // <9>
+
+        return authors;
+    }
+----
+<1> Important point: we need a transactional context for these methods.
```

Comment:
You need a transaction to run the mass indexer? That's weird... ?

Reply:
Yes, otherwise you end up with:
```
javax.enterprise.context.ContextNotActiveException: interface javax.enterprise.context.RequestScoped
```
it's specific to Quarkus though.